packages/graphql/src/contracts.js

```diff
@@ -62,16 +63,23 @@ export function newBlock(blockHeaders) {
 }
 
 function pollForBlocks() {
+  let inProgress = false
   try {
     blockInterval = setInterval(() => {
+      if (inProgress) {
+        return
+      }
+      inProgress = true
       web3.eth.getBlockNumber().then(block => {
         if (block > lastBlock) {
           web3.eth.getBlock(block).then(newBlock)
         }
+        inProgress = false
       })
     }, 5000)
   } catch (error) {
     console.log(`Polling for new blocks failed: ${error}`)
+    inProgress = false
```

Comment:
Should we throw an exception here ? Since it basically means new blocks won't get polled at all.
Seems a critical error. Luckily should never happen since would only occur if for some reason setInterval fails ?

Reply:
hm, I hadn't considered this when I was looking at it.  Does `setInterval()` die forever on an error?  What if we moved the try/catch into the setInterval function argument?